src/gluonnlp/models/electra.py

```diff
@@ -337,6 +347,58 @@ def get_initial_embedding(self, F, inputs, token_types=None):
         embedding = self.embed_dropout(embedding)
         return embedding
 
+    def apply_layerwise_decay(self, layerwise_decay, not_included=[]):
+        """Apply the layer-wise gradient decay
+
+        .. math::
+            lr = lr * layerwise_decay^(max_depth - layer_depth)
+
+        Parameters:
+        ----------
+        layerwise_decay: int
+            layer-wise decay power
+        not_included: list of str
+            A list or parameter names that not included in the layer-wise decay
+        """
+
+        # consider the task specific finetuning layer as the last layer, following with pooler
+        # In addition, the embedding parameters have the smaller learning rate based on this setting.
+        max_depth = self.num_layers
+        for _, value in self.collect_params('.*embed*').items():
+            value.lr_mult = layerwise_decay**(max_depth + 1)
```

Comment:
For me, I find that I need to use `max_depth = self.num_layers + 2` for text classification: https://github.com/sxjscience/gluon-nlp/blob/adcd22301f73ea05aa366bbb41c3016f3aac5186/src/gluonnlp/auto/estimators/basic_v1.py#L93-L108